### ISSUE TITLE
Sugestion : Select In filter.

### ISSUE THREAD

**Issue by hfmiguel** [lacodix/laravel-model-filter#46]
Hi , 

I think that an **select in** filter will be a good improvement, many situations depends that we search in multiples values .

At this moment, i write this *Base Filter*  to use as Select In filter and its workin like a charm. 

Maybe can be improved and implemented in a future release ? 

```
<?php

namespace App\Models\Tenant\Filters\vendor;

use Illuminate\Database\Eloquent\Builder;
use Illuminate\Support\Arr;
use Lacodix\LaravelModelFilter\Filters\Filter;

class SelectInFilter extends Filter
{
    protected string $field;

    public function apply(Builder $query): Builder
    {
        if (is_array($this->values)) {
            return $query->whereIn($this->field, $this->values);
        } else {
            return $query->where($this->field, $this->values);
        }
    }

    public function populate(string|array $values): self
    {
        $this->values = Arr::wrap($values);
        return $this;
    }
}
```

**Comment by renky** [comment]
I'm not shure if I understood the usecase correctly: On the Webpage you want to select one or multiple values, and the filter then shall find all entries that belongs to at least one of it? That's what I expect from your example...

The question for me is: how will this filter be visualized? In my imagination I'd expect a set of Checkboxes, where I can select one or multiple checkboxes, and the filter then searches for one or multiple of this values....
This can already be realized with the BooleanFilter - maybe the name isn't that good - we could also call it CheckboxFilter - but in the end it is a filter that can show you multiple values, where you can select one or more of it...

Please have a look into "Calculated Values" Section of the Boolean filter... you can just use your apply function there I quess...
https://www.lacodix.de/docs/laravel-model-filter/v1/filter-types/boolean#content-calculated-values
You are right, that a dedicated SelectedIn Filter would save you the time of overwriting the apply-method...

Btw: in your example you don't need to check for "is_array" in the apply method - since you use Arr::wrap in populate function you have always an array and can remove the if completely....

**Comment by hfmiguel** [comment]
Yeah , thats right !

Im building my own views components , and the visualization will be : 

![image](https://github.com/lacodix/laravel-model-filter/assets/14097051/f5bfb1f0-87c8-4746-a1ef-49315636004b)

![image](https://github.com/lacodix/laravel-model-filter/assets/14097051/8fbd39a0-9ac4-459e-ae4c-7245addf8089)


At this moment  , the component works fine (visually and search ) . 


The only things missing its a default filter to receive an array of data and apply the filter . 

With the example that i sent , the filter work fine .


And thanks , i modify the filter to : 

```
 public function apply(Builder $query): Builder
    {
        return $query->whereIn($this->field, $this->values);
    }

    public function populate(string|array $values): self
    {
        $this->values = Arr::wrap($values);
        return $this;
    }
```
More clean. 


Will be nice in a future release a default filter to filter an array of data.

**Comment by renky** [comment]
Ok, got it...

you can even remove the populate function completly, because yours is identical now with the base-populate-function... 

with that in mind it is a really "simple" functionality - just an apply-function with a whereIn...
but I think your component will not fit into the package, because such a component is very css and js-related... I try to keep all the filters and there visualisations headless - just using css-classes so everybody can style it on it's own... offering such a component would not be visualisable with that "light" approach...

so I need a visualisation without dropdown and checkboxes in this dropdown...

maybe it could be realized in visualisation like the current Booleanfilter (just a set of checkboxes - without dropdown - every checkbox has its own array-key and value, and it will have two modes: one selected value must fit, or all selected values...
I've already created such a filter in one of my projects and called it "multiselect" - but since this also uses a separate js-library (Select2.js) it didn't find its way to this package...

**Comment by hfmiguel** [comment]
On really, dont need a checkbox . 

If in the package components , the user could define the select with the **multiple** attribute ,so ,  the html will send an array of data.

Dont need any additional configuration .

**Comment by renky** [comment]
You're right - I totally ignored the multi-select-option^^ Yeah, thats an option, and since the default select-filter doesn't have a "mode" yet, it could be added and part of the normal select-filter... and if you want to use your own component, you can just overwrite the component for that...

**Comment by renky** [comment]
Hey @hfmiguel 

it took a long time but I just released a new Version v1.11.x that also contains the Select-in-Filter
You can now use the Filtermode CONTAINS for select filter which will result in a multiple-select-component, and searches for array-containing.

see more details here
https://www.lacodix.de/docs/laravel-model-filter/v1/filter-types/select#content-multiselect

**Comment by hfmiguel** [comment]
Good, 

Oops, thanks for the feedback. I'll try updating the package in my next release to see. 

Thank you
